### DISCUSSION TITLE
Training on remote machine, inference locally

### DISCUSSION THREAD

**Discussion by agosztolai** [talmolab/sleap#816] [Help!] [answered]
Hello, 

I have exported my human-labelled images by clicking 'Export training job package', uploaded them to my remote machine and then launched 'sleap-train single_instance.json labels.v000.pkg.slp'. Now I have a trained model.

Since I have selected 'Predict on: entire current video' at the export stage, I have expected the labels of that video to appear somewhere, which I would need to refine. However, I am not sure where these predictions are.

Now that I have trained, I want to move my files (models, labels etc) back to my laptop to refine labels and then push bash them back to the remote machine for another round of training.

After I move my files back locally, which file do I need to open that contains the refined labels? Do I need to do any merging of datasets before I push to the remove machine for another round of training?

Thanks!

**Comment by roomrys** [comment, marked as answer]
Hi @agosztolai,

If you ran inference via the GUI, then you do not need to merge predictions into the project - the predictions will have been merged for you. However, if you ran inference via the CLI, then you will need to open your original project and [import the predictions](https://sleap.ai/guides/merging.html) into the original via File > Merge Into Project...

**To merge predictions (if inference was run via CLI)**
If you ran the [`sleap-track`](https://sleap.ai/guides/cli.html#sleap-track) command, the predictions will have been saved to the output filename specified by the optional `-o` or `--output` argument. If no output path was specified, the output defaults to `[data_path].predictions.slp` where `data_path` is the (required) positional argument used in `sleap-track`.

**Locating predictions - no need to merge (if inference was run via GUI)**
After running inference, you should have been left with a `./predictions/` directory which contains a list of `.predictions.slp` files which are made up of only predictions. The naming convention for the `.predictions.slp` is `<original project name><model id>.predictions.slp`; thus, to locate the correct `.predictions.slp`, you just need to know the corresponding project name and model id used to run inference (all models are located in the `./models/` folder by default).

Let me know if that answered your question or if anything needs to be clarified. Happy to help!

```
│   labels.v000.slp
│   mice.mp4
│
├───models
│   └───220404_141706.centered_instance.n=15
│           best_model.h5
│           initial_config.json
│           labels_gt.train.slp
│           labels_gt.val.slp
│           labels_pr.train.slp
│           labels_pr.val.slp
│           metrics.train.npz
│           metrics.val.npz
│           training_config.json
│           training_log.csv
│
└───predictions
        labels.v000.slp.220404_141706.predictions.slp
```
**Fig 1**: An example directory tree showcasing where the `./models/` and `./predictions/` folders are located w.r.t. the original SLEAP Project file (if inference is run via the GUI).

Thanks,
Liezl

**Reply by agosztolai** [reply]
Thank you! This solved my problem.
